src/btree/bt_split.c

```diff
@@ -972,6 +980,9 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF **ref_new,
 		parent_decr += sizeof(WT_REF);
 	}
 
+	/* We freed the reference that was split in the loop above. */
+	ref = NULL;
+
```

Comment:
That said, we can just make that clear in comments, rather than an assignment that lint objects to.